### PR TITLE
Bug 1256261 - [TV][Home] Add and modify new folder through grid-view. r=lchang

### PR DESCRIPTION
**Pull request by rexboy7**
This patch implements basic feature of picking folder from grid-view dialog.

There are TODOs remaining:
1. Limit number of folder to 9.
2. Visual spec.
3. Sort cards (except thoses inside a folder) by alphabet order in card-picker.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fc23540045aaacfa748557f201f19d931fac8844

File: shared/js/smart-screen/key_navigation_adapter.js [hunk 1]
```diff
@@ -44,6 +45,23 @@
...
+        case 'keydown':
+        case 'keypress':
+          // By default we stop propagating keyevent. This is useful when we
+          // apply KeyNavigationAdapter to a blocked sub-component such as a
+          // modal-dialog-like UI. It prevents underlying UI act to the same
+          // key.
+          !this.propagateKeyEvent && evt.stopPropagation();
+          break;
+        case 'click':
+        // Do forced focus again on mouse click to prevent focus lost problem
```
Comment (luke-chang) [inline shared/js/smart-screen/key_navigation_adapter.js]: wrong indention.

File: tv_apps/smart-home/js/card_picker.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+(function(exports) {
+  'use strict';
+
+  function CardPicker() {}
+
+  CardPicker.prototype = evt({
+    container: document.getElementById('card-picker'),
+    gridView: document.getElementById('card-picker-grid-view'),
+
+    hideCardpickerButton: document.getElementById('hide-cardpicker-button'),
```
Comment (luke-chang) [inline tv_apps/smart-home/js/card_picker.js]: Isn't that `hideCardPickerButton`?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=76944190966b25193c2025445cce287f77b000f7

**Pull request merged**